indexing-service/src/main/java/org/apache/druid/indexing/worker/WorkerTaskManager.java

```diff
@@ -184,6 +184,20 @@ public Map<String, TaskAnnouncement> getCompletedTasks()
     return completedTasks;
   }
 
+  public Map<String, TaskAnnouncement> getFailedTasks()
```

Comment:
why not return the list/set of task announcements itself since you only need that. This method can be just combined with getWorkedFailedTasks.